src/aggregate/aggregate_exec.c

```diff
@@ -205,17 +205,20 @@ static size_t serializeResult(AREQ *req, RedisModule_Reply *reply, const SearchR
         // For duo value, we use the left value here (not the right value)
         v = RSValue_Trio_GetLeft(v);
       }
-      RSValue rsv;
+      RSValue *rsv = NULL;
       if (rlk && (rlk->flags & RLOOKUP_T_NUMERIC) && v && !RSValue_IsNumber(v) && !RSValue_IsNull(v)) {
         double d;
         RSValue_ToNumber(v, &d);
-        RSValue_SetNumber(&rsv, d);
-        v = &rsv;
+        rsv = RSValue_NewNumber(d);
```

Comment:
We are allocating the variable inside the loop, can't we allocate it out of the loop and reuse the variable? or maybe use lazy allocation on first use?